internal/serverinstall/k8s.go

```diff
@@ -711,189 +712,235 @@ func (i *K8sInstaller) Uninstall(ctx context.Context, opts *InstallOpts) error {
 		return err
 	}
 
-	s.Update("Deleting any automatically installed runners...")
-
-	// create our wait channel to later poll for statefulset+pod deletion
-	w, err := clientset.AppsV1().Deployments(i.config.namespace).Watch(
-		ctx,
-		metav1.ListOptions{
-			LabelSelector: "app=" + runnerName,
-		},
-	)
-
-	// send DELETE to statefulset collection
-	if err = clientset.AppsV1().Deployments(i.config.namespace).DeleteCollection(
-		ctx,
-		metav1.DeleteOptions{},
-		metav1.ListOptions{
-			LabelSelector: "app=" + runnerName,
-		},
-	); err != nil {
+	deploymentClient := clientset.AppsV1().Deployments(i.config.namespace)
+	if list, err := deploymentClient.List(ctx, metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("app=%s", runnerName),
+	}); err != nil {
 		ui.Output(
-			"Error deleting Waypoint deployment: %s", clierrors.Humanize(err),
+			"Error looking up deployments: %s", clierrors.Humanize(err),
 			terminal.WithErrorStyle(),
 		)
 		return err
-	}
+	} else if len(list.Items) > 0 {
+		s.Update("Deleting any automatically installed runners...")
+
+		// create our wait channel to later poll for statefulset+pod deletion
+		w, err := deploymentClient.Watch(
+			ctx,
+			metav1.ListOptions{
+				LabelSelector: "app=" + runnerName,
+			},
+		)
 
-	// wait for deletion to complete
-	err = wait.PollImmediate(2*time.Second, 10*time.Minute, func() (bool, error) {
-		select {
-		case wCh := <-w.ResultChan():
-			if wCh.Type == "DELETED" {
-				w.Stop()
-				return true, nil
+		// send DELETE to statefulset collection
+		if err = deploymentClient.DeleteCollection(
+			ctx,
+			metav1.DeleteOptions{},
+			metav1.ListOptions{
+				LabelSelector: "app=" + runnerName,
+			},
+		); err != nil {
+			ui.Output(
+				"Error deleting Waypoint deployment: %s", clierrors.Humanize(err),
+				terminal.WithErrorStyle(),
+			)
+			return err
+		}
+
+		// wait for deletion to complete
+		err = wait.PollImmediate(2*time.Second, 10*time.Minute, func() (bool, error) {
```

Comment:
I've noticed we end up waiting 10 minutes for a lot of things in the k8s platform. Is this a reasonable amount of time? Is 5 minutes not enough?

Reply:
On Azure stuff like PVCs sometimes take that long. :|